File: problems/1317_convert_integer_to_the_sum_of_two_no-zero_integers/1317_convert_integer_to_the_sum_of_two_no-zero_integers.java
class Solution {
    public int[] getNoZeroIntegers(int n) {
        for (int i = 1; i < n; i++) {
            if (!hasZero(i) && !hasZero(n - i)) {
                return new int[] { i, n - i };
            }
        }
        return new int[] { -1, -1 };
    }

    public boolean hasZero(int n) {
        while (n > 0) {
            if (n % 10 == 0) {
                return true;
            }
            n /= 10;
        }
        return false;
    }
}

File: problems/1317_convert_integer_to_the_sum_of_two_no-zero_integers/1317_convert_integer_to_the_sum_of_two_no-zero_integers.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        def has_zero(x):
            while x > 0:
                if x % 10 == 0:
                    return True
                x //= 10
            return False
        
        for i in range(1, n):
            if not has_zero(i) and not has_zero(n - i):
                return [i, n - i]
        return [-1, -1]